File: LeetCode/1100-1500/1500/1582/LC1582SpecialPositions.js




{
    // way 01
    // brute force 


    /**
     * @param {number[][]} mat
     * @return {number}
   */
  
    var numSpecial = function (mat) {

        let count = 0;

        for (let i = 0; i < mat.length; i++) {

            for (let j = 0; j < mat[0].length; j++) {

                if (mat[i][j] === 1 && isSpecial(i, j, mat)) {
                    count++;
                }
            }
        }

        return count;

    };

    function isSpecial(row, col, mat) {

        for (let i = 0; i < mat.length; i++) {

            if (mat[i][col] !== 0 && i !== row) {
                return false;
            }
        }

        for (let j = 0; j < mat[0].length; j++) {

            if (mat[row][j] !== 0 && j !== col) {
                return false;
            }
        }

        return true;
    }
}

File: DSA/Leetcode/LeetcodeProblem/1500/1582/LC1582SpecialPositions.java
class Solution {
    public int numSpecial(int[][] mat) {
        int count = 0;

        for (int i = 0; i < mat.length; i++){
            for (int j = 0; j < mat[i].length; j++){
                if (mat[i][j] == 1 && isSpecial(i, j, mat)){
                    count++;
                }
            }
        }
        return count;
    }

   public boolean isSpecial(int i, int j, int[][] mat) {
    for (int k = 0; k < mat.length; k++) {
        if (k == i) {
            continue;
        }
        if (mat[k][j] == 1) {
            return false;
        }
    }